rtos/Mutex.h

```diff
@@ -100,8 +97,9 @@ class Mutex : private mbed::NonCopyable<Mutex> {
                @a osErrorISR this function cannot be called from the interrupt service routine.
 
       @note You cannot call this function from ISR context.
+      @note This function asserts status of the lock operation (will not return in case of failure).
```

Comment:
Sorry, you've gotten confused. `lock(millisec)` is wanted to be deprecated.

The change was to put back the `osStatus` on `lock()`, but say that use of it was deprecated. So this green bit goes onto `lock()`, and put back the deprecation on `lock(millisec)`.